src/prefect/task_engine.py

```diff
@@ -369,12 +373,58 @@ def key_fn():
         return result
 
     def handle_retry(self, exc: Exception) -> bool:
+        """Handle a task run retry.
+
+        - If the task has retries left, and the retry condition is met, set the task to retrying.
+            - If the task has a `retry_delay_seconds` set, wait for that amount of time before retrying.
+            - If we have more retries than `retry_delay_seconds` values, use the last value in the sequence.
+        - If the task has no retries left, or the retry condition is not met, return False.
+
         """
-        If the task has retries left, and the retry condition is met, set the task to retrying.
+        if self.retries < self.task.retries and self.can_retry:
+            if self.task.retry_delay_seconds:
+                if isinstance(self.task.retry_delay_seconds, Sequence):
+                    index = min(self.retries, len(self.task.retry_delay_seconds) - 1)
+                    delay = self.task.retry_delay_seconds[index]
+                else:
+                    delay = self.task.retry_delay_seconds
+
+                if isinstance(delay, (int, float)):
+                    self.logger.info(f"Waiting {delay} seconds before retrying...")
+                    time.sleep(delay)
+                else:
+                    raise ValueError(
+                        "Invalid type for `retry_delay_seconds`. Must be an int, float, or Sequence[Union[int, float]]"
+                    )
+            self.set_state(Retrying(), force=True)
+            self.retries = self.retries + 1
+            return True
+        return False
+
+    async def handle_retry_async(self, exc: Exception) -> bool:
```

Comment:
This would be the first place in the engine where we use an async engine method; i get why you did it of course, but I'd like to consider other ways we might do this